cdc/cdc.cc

```diff
@@ -218,4 +219,195 @@ db_context db_context::builder::build() {
     };
 }
 
+class transformer final {
+public:
+    struct map_cmp {
+        template <typename Func>
+        bool cmp(const std::pair<net::inet_address, unsigned int>& a,
+                 const std::pair<net::inet_address, unsigned int>& b,
+                 Func&& func) const {
+            auto&& a_ip = func(a.first).ip;
+            auto&& b_ip = func(b.first).ip;
+            return a_ip < b_ip || (a_ip == b_ip && a.second < b.second);
+        }
+        bool operator()(const std::pair<net::inet_address, unsigned int>& a,
+                        const std::pair<net::inet_address, unsigned int>& b) const {
+            if (a.first.is_ipv4()) {
+                return !b.first.is_ipv4() ||
+                    cmp(a, b, [] (const net::inet_address& addr) { return addr.as_ipv4_address(); });
+            } else {
+                return !b.first.is_ipv4() &&
+                    cmp(a, b, [] (const net::inet_address& addr) { return addr.as_ipv6_address(); });
+            }
+        }
+    };
+    using streams_type = std::map<std::pair<net::inet_address, unsigned int>, utils::UUID, map_cmp>;
+private:
+    db_context _ctx;
+    const schema& _schema;
+    schema_ptr _log_schema;
+    utils::UUID _time;
+    bytes _decomposed_time;
+    // The assumption here is that all nodes in the cluster are using the
+    // same murmur3_partitioner_ignore_msb_bits. If it's not true, we will
+    // have to gossip it around together with shard_count.
+    unsigned _ignore_msb_bits;
```

Comment:
We'll need a map here as the assumption is incorrect.

Reply:
@haaawk maybe need to add a FIXME here? Or you're just assuming that everything around this area will be rewritten in the near future anyway?

Reply:
I fixed the issue by gossiping the ignore_msb_bits for each node and using the right value when needed.